src/engraving/libmscore/slur.cpp

```diff
@@ -377,10 +377,37 @@ void SlurSegment::avoidCollisions(PointF& pp1, PointF& p2, PointF& p3, PointF& p
 {
     ChordRest* startCR = slur()->startCR();
     ChordRest* endCR = slur()->endCR();
-    Segment* startSeg = startCR ? startCR->segment() : nullptr;
-    Segment* endSeg = endCR ? endCR->segment() : nullptr;
     bool isCrossStaff = (startCR && endCR) ? startCR->vStaffIdx() != endCR->vStaffIdx() : false;
 
+    Segment* startSeg = nullptr;
+    if (isSingleBeginType()) {
+        startSeg = startCR ? startCR->segment() : nullptr; // first of the slur
+    } else {
+        startSeg = system()->firstMeasure()->findFirstR(SegmentType::ChordRest, Fraction(0, 0)); // first of the system
+    }
+    Segment* endSeg = nullptr;
+    if (isSingleEndType()) {
+        endSeg = endCR ? endCR->segment() : nullptr; // last of the slur
+    } else {
+        endSeg = system()->lastMeasure()->last(); // last of the system
+    }
+
+    // Collect all the segments spanned by this slur segment in a single vector
+    std::vector<Segment*> segList;
+    Segment* seg = startSeg;
+    while (seg && seg->tick() <= endSeg->tick()) {
```

Comment:
Looks like this will go wrong if `endSeg` is `nullptr`. It looks like if `startSeg` or `endSeg` is `nullptr`, the code below is not executed anyway. Maybe it makes sense to add something like the following here above?
```c++
if (!startSeg || !endSeg) { // maybe also `|| startSeg == endSeg`?
    return;
}
```

Also, maybe in the `if` statement in line 417, you could replace `startSeg && endSeg && startSeg != endSeg` with `segList.empty()`.